src/connector/sqlite/conversion.rs

```diff
@@ -17,56 +17,48 @@ use chrono::TimeZone;
 
 impl TypeIdentifier for Column<'_> {
     fn is_real(&self) -> bool {
-        match self.decl_type() {
-            Some(n) if n.starts_with("DECIMAL") => true,
+        match self.decl_type().map(str::to_lowercase).as_deref() {
             Some(n) if n.starts_with("decimal") => true,
-            Some("NUMERIC") | Some("REAL") => true,
             Some("numeric") | Some("real") => true,
             _ => false,
         }
     }
 
     fn is_float(&self) -> bool {
-        matches!(self.decl_type(), Some("float") | Some("FLOAT"))
+        matches!(self.decl_type().map(str::to_lowercase).as_deref(), Some("float"))
     }
 
     fn is_double(&self) -> bool {
         matches!(
-            self.decl_type(),
-            Some("double") | Some("DOUBLE") | Some("DOUBLE PRECISION") | Some("double precision")
+            self.decl_type().map(str::to_lowercase).as_deref(),
```

Comment:
Same here... Either find a way to compare strings case-insensitive, or compare all cases. Silent allocation in a method like this is not cool.